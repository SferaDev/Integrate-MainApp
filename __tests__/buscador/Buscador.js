import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Buscador from '../../components/buscador/buscador';

const navigation = {navigate: jest.fn()};
let wrapper;
let instance;

describe('Test group for EntityList', function () {
    beforeAll(() => {
        jest.mock('react-native-maps', () => require.requireActual('../../__mocks__/react-native-maps').default);
        configure({adapter: new Adapter()});
    });

    beforeEach(function () {
        // Before each: Shallows the EntityList component
        wrapper = shallow(<Buscador navigation={navigation}/>);
        instance = wrapper.instance();
    });

    afterEach(function () {
        // After each: Clears the wrapper
        wrapper = null;
        instance = null;
    });

    test('renders buscador correctly', () => {
        const tree = renderer.create(<Buscador navigation={navigation}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('openMenu is callable and returns nothing', () => {
        expect(instance.openMenu()).toBe(undefined);
    });

    test('switchView is callable and returns nothing', () => {
        expect(instance.switchView()).toBe(undefined);
    });

    test('getEntities is callable and returns an array', () => {
        expect(instance.getEntities()).toBe(undefined);
    });

    test('setEntities is callable and returns an array', () => {
        expect(instance.setEntities()).toBe(undefined);
    });

    test('handleBackButton is callable and returns true', () => {
        expect(instance.handleBackButton()).toBe(true);
    });

    test('showEntityInfo is callable and returns nothing', () => {
        expect(instance.showEntityInfo()).toBe(undefined);
    });

    test('setEntities is callable and returns nothing', () => {
        expect(instance.setEntities()).toBe(undefined);
    });

    test('showListView is callable and returns nothing', () => {
        expect(instance.showListView()).toBe(undefined);
    });

    test('updateSearchText is callable and returns nothing', () => {
        expect(instance.updateSearchText()).toBe(undefined);
    });

    test('filterEntities is callable and returns nothing', () => {
        expect(instance.filterEntities()).toBe(undefined);
    });

    test('filterEntities where searchText is UPC is callable and returns nothing', () => {
        instance.state.searchText = "UPC";
        expect(instance.filterEntities()).toBe(undefined);
    });

    test('filterEntities where entities are UPC and Patates Jordi is callable and returns nothing', () => {
        instance.state.searchText = "UP";
        instance.state.entities = [{name: "UPC"}, {name: "Patates Jordi"}];
        expect(instance.filterEntities()).toBe(undefined);
    });

    describe("isAnEntitySelected() tests", () => {
        it('isAnEntitySelected() when NifNie and Password are empty then returns true', () => {

            instance.state.selectedEntity = null;
            expect(instance.isAnEntitySelected()).toBe(false);
        });

        it('isAnEntitySelected() when NifNie is empty and Password is filled then returns true', () => {

            instance.state.selectedEntity = {
                id: 1,
                name: 'name',
                description: 'description',
                addressName: 'addressName',
            };
            expect(instance.isAnEntitySelected()).toBe(true);
        });
    });

    describe("isListView() tests", () => {
        it('isListView() when NifNie and Password are empty then returns true', () => {

            instance.state.isListView = true;
            expect(instance.isListView()).toBe(0);
        });

        it('isListView() when NifNie is empty and Password is filled then returns true', () => {

            instance.state.isListView = false;
            expect(instance.isListView()).toBe(100);
        });
    });
});
