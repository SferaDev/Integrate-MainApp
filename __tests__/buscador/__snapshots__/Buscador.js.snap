// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test group for EntityList renders buscador correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#094671",
        "display": "flex",
        "flexDirection": "row",
        "height": 60,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "flex": 1,
          "paddingLeft": 20,
          "textAlign": "left",
        }
      }
    >
      MENU
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "flex": 1,
          "paddingRight": 20,
          "textAlign": "right",
        }
      }
    >
      LIST
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 8,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        followsUserLocation={true}
        initialRegion={
          Object {
            "latitude": 0,
            "latitudeDelta": 0.0922,
            "longitude": 0,
            "longitudeDelta": 0.0421,
          }
        }
        showsCompass={false}
        showsMyLocationButton={false}
        showsPointsOfInterest={false}
        showsUserLocation={true}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "paddingTop": 100,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "red",
              "borderRadius": 15,
              "bottom": 15,
              "height": 30,
              "left": 15,
              "position": "absolute",
              "width": 30,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          NO
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "red",
              "borderRadius": 22.5,
              "bottom": 15,
              "height": 45,
              "position": "absolute",
              "right": 15,
              "width": 45,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          ME
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 100,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 37,
        "position": "absolute",
        "top": 75,
        "width": 345,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "flex": 2,
          "textAlign": "center",
        }
      }
    >
      Q
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={false}
      placeholder="Search"
      style={
        Object {
          "flex": 18,
        }
      }
    />
  </View>
</View>
`;
